### DISCUSSION TITLE
Change color of border on CheckBox

### DISCUSSION THREAD

**Discussion by roxxel** [AvaloniaUI/Avalonia#6679] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/64275777/136222631-5913223d-4f3e-4fbe-b92f-4dd89730e17b.png)
How can i change color of border on checkbox
(using BorderBrush i can change color, but that's not what i need)

**Comment by MarkusKgit** [comment, marked as answer]
Just an example (using Fluent Theme):
```xaml
<Grid>
  <Grid.Styles>
    <Style Selector="CheckBox /template/ Border#NormalRectangle">
      <Setter Property="BorderBrush" Value="Red"/>        
    </Style>
    <Style Selector="CheckBox:pointerover /template/ Border#NormalRectangle">
      <Setter Property="BorderBrush" Value="Green" />
    </Style>
  </Grid.Styles>
  <CheckBox Content="Check me"/>
</Grid>
```
![CB](https://user-images.githubusercontent.com/13160892/136528248-2483bae3-2eee-4188-a6c8-4c03a73d370c.png) ![CB2](https://user-images.githubusercontent.com/13160892/136530102-e4ee4db5-0b1a-4438-a081-a6b7474b2155.png)

In addition to `:pointerover` you might want to also handle `:pressed`, `:disabled`, `:checked`, `:checked:pointerover`, `:checked:pressed` and `:checked:disabled`

**Reply by mikhailSokolovskiy** [reply]
how do I access the third state of the checkbox? when selected any itmes, like a rectangle instead tick

**Reply by thevortexcloud** [reply]
Check the original styles. I am guessing it's this: 
https://github.com/AvaloniaUI/Avalonia/blob/fd15b6bc9e07135472b5e2ee12bc8fba7d7f5567/src/Avalonia.Themes.Fluent/Controls/CheckBox.xaml#L218-L234

**Reply by VoidPlayz510** [reply]
I found doing this worked

```xaml
<Style Selector="CheckBox:unchecked /template/ Border#NormalRectangle">
	      <Setter Property="BorderBrush" Value="Red"/>
</Style>
 <Style Selector="CheckBox:pointerover /template/ Border#NormalRectangle">
	<Setter Property="BorderBrush" Value="Green" />
</Style>
```
